x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/FieldPermissions.java

```diff
@@ -61,9 +61,8 @@ public final class FieldPermissions implements Accountable, CacheKey {
         BASE_HASHSET_ENTRY_SIZE = mapEntryShallowSize + 2 * RamUsageEstimator.NUM_BYTES_OBJECT_REF;
     }
 
-    private final FieldPermissionsDefinition fieldPermissionsDefinition;
-    @Nullable
-    private final FieldPermissionsDefinition limitedByFieldPermissionsDefinition;
+    private final List<FieldPermissionsDefinition> fieldPermissionsDefinitions;
```

Comment:
This is the critical change of this PR: replacing two hard coded variables with a List (similar to that of #91151). Everything else revolves around it.